src/test/java/net/kaczmarzyk/InE2eTest.java

```diff
@@ -110,111 +126,151 @@ public void findsByListOfAllowedStringValues() throws Exception {
 		mockMvc.perform(get("/customers")
 				.param("firstNameIn", "Homer", "Marge")
 				.accept(MediaType.APPLICATION_JSON))
-				.andExpect(status().isOk())
-				.andExpect(jsonPath("$").isArray())
-				.andExpect(jsonPath("$[0].firstName").value("Homer"))
-				.andExpect(jsonPath("$[1].firstName").value("Marge"))
-				.andExpect(jsonPath("$[2]").doesNotExist());
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+			.andExpect(jsonPath("$[0].firstName").value("Homer"))
+			.andExpect(jsonPath("$[1].firstName").value("Marge"))
+			.andExpect(jsonPath("$[2]").doesNotExist());
 	}
 
 	@Test
 	public void findsByListOfAllowedStringValuesUsingParamSeparator() throws Exception {
-		mockMvc.perform(get("/customers-ps?firstNameIn=Homer,Marge")
+		mockMvc.perform(get("/customers-param-separator?firstNameIn=Homer&firstNameIn=Moe,Bart&firstNameIn=Lisa,Maggie")
 				.accept(MediaType.APPLICATION_JSON))
-				.andExpect(status().isOk())
-				.andExpect(jsonPath("$").isArray())
-				.andExpect(jsonPath("$[0].firstName").value("Homer"))
-				.andExpect(jsonPath("$[1].firstName").value("Marge"))
-				.andExpect(jsonPath("$[2]").doesNotExist());
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+			.andExpect(jsonPath("$[0].firstName").value("Homer"))
+			.andExpect(jsonPath("$[1].firstName").value("Bart"))
+			.andExpect(jsonPath("$[2].firstName").value("Lisa"))
+			.andExpect(jsonPath("$[3].firstName").value("Maggie"))
+			.andExpect(jsonPath("$[4].firstName").value("Moe"))
+			.andExpect(jsonPath("$[5]").doesNotExist());
+	}
+
+	@Test
+	public void findsByListOfAllowedStringValuesUsingWebParameterNameTheSameAsPathName() throws Exception {
+		mockMvc.perform(get("/customers-with-missing-parameter-name")
+				.param("firstName", "Homer", "Marge")
+				.accept(MediaType.APPLICATION_JSON))
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+			.andExpect(jsonPath("$[0].firstName").value("Homer"))
+			.andExpect(jsonPath("$[1].firstName").value("Marge"))
+			.andExpect(jsonPath("$[2]").doesNotExist());
+	}
+
+	@Test
+	public void findsByListOfAllowedStringValuesUsingWebParameterNameTheSameAsPathNameWithoutUsingParamSeparator() throws Exception {
+		mockMvc.perform(get("/customers-with-missing-parameter-name?firstName=Homer&firstName=Moe,Bart&firstName=Lisa,Maggie")
+				.accept(MediaType.APPLICATION_JSON))
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+			.andExpect(jsonPath("$[0].firstName").value("Homer"))
```

Comment:
I don't get it why only Homer is here. If delimitation is disabled, I would expect these values: "Homer", "Moe,Bart", "Lisa,Maggie" 

??

Reply:
because there are not any customer with firstName `Moe,Bart` or `Lisa,Maggie` does not exist